consensus/babe/runtime.go

```diff
@@ -46,34 +46,55 @@ func (b *Session) configurationFromRuntime() error {
 
 // calls runtime API function Core_initialize_block
 func (b *Session) initializeBlock(blockHeader []byte) error {
-	// TODO: use allocator to store block header
-	var loc int32 = 1
-	b.rt.Store(blockHeader, loc)
+	ptr, err := b.rt.Malloc(uint32(len(blockHeader)))
+	if err != nil {
+		return err
+	}
+
+	b.rt.Store(blockHeader, int32(ptr))
 
-	_, err := b.rt.Exec(runtime.CoreInitializeBlock, loc, blockHeader)
+	_, err = b.rt.Exec(runtime.CoreInitializeBlock, int32(ptr), blockHeader)
 	if err != nil {
 		return err
 	}
 
-	return nil
+	return b.rt.Free(ptr)
 }
 
 // calls runtime API function BlockBuilder_inherent_extrinsics
-func (b *Session) inherentExtrinsics(blockInherentData []byte) ([]byte, error) { //nolint:unused
-	// TODO: use allocator to store inherents data
-	var loc int32 = 1
-	b.rt.Store(blockInherentData, loc)
+func (b *Session) inherentExtrinsics(data []byte) ([]byte, error) { //nolint:unused
+	ptr, err := b.rt.Malloc(uint32(len(data)))
+	if err != nil {
+		return nil, err
+	}
 
-	return b.rt.Exec(runtime.BlockBuilderInherentExtrinsics, loc, blockInherentData)
+	b.rt.Store(data, int32(ptr))
+
+	ret, err := b.rt.Exec(runtime.BlockBuilderInherentExtrinsics, int32(ptr), data)
```

Comment:
Is `ret` short for return? We should be more descriptive here and elsewhere (ie, `rawdata`).